gammapy/modeling/models/spatial.py

```diff
@@ -1289,6 +1290,104 @@ def plot_interactive(self, ax=None, geom=None, **kwargs):
 
 
+class TemplateNDSpatialModel(SpatialModel):
+    """A model generated from a ND map where extra dimensions define the parameter space.
+
+    Parameters
+    ----------
+    map : `~gammapy.maps.WcsNDMap`
+        Map template.
+    meta : dict, optional
+        Meta information, meta['filename'] will be used for serialization
+    interp_kwargs : dict
+        Interpolation keyword arguments passed to `gammapy.maps.Map.interp_by_pix`.
+        Default arguments are {'method': 'linear', 'fill_value': 0, "values_scale": "log"}.
+    """
+
+    tag = ["TemplateNDSpatialModel", "templateND"]
+
+    def __init__(self, map, interp_kwargs=None, meta=None, filename=None):
+        self._map = map.copy()
+        self.meta = dict() if meta is None else meta
+        if filename is not None:
+            filename = str(make_path(filename))
+        self.filename = filename
+
+        parameters = []
+        for axis in map.geom.axes:
+            if axis.name not in ["energy_true", "energy"]:
+                center = (axis.bounds[1] + axis.bounds[0]) / 2
+                parameter = Parameter(
+                    name=axis.name,
+                    value=center,
+                    unit=axis.unit,
+                    scale_method="scale10",
+                    min=axis.bounds[0],
+                    max=axis.bounds[-1],
+                    interp=axis.interp,
+                )
+                parameters.append(parameter)
+        self.default_parameters = Parameters(parameters)
+
+        interp_kwargs = interp_kwargs or {}
+        interp_kwargs.setdefault("method", "linear")
+        interp_kwargs.setdefault("fill_value", 0)
+        interp_kwargs.setdefault("values_scale", "log")
+        self._interp_kwargs = interp_kwargs
+        super().__init__()
+
+    @property
+    def map(self):
+        """Template map  (`~gammapy.maps.WcsNDMap`)"""
+        return self._map
+
+    @property
+    def is_energy_dependent(self):
+        return "energy_true" in self.map.geom.axes.names
+
+    def evaluate(self, lon, lat, energy=None, **kwargs):
+        coord = {
+            "lon": lon.to_value("deg"),
+            "lat": lat.to_value("deg"),
+        }
+        if energy is not None:
+            coord["energy_true"] = energy
+
+        coord.update(kwargs)
+
+        val = self.map.interp_by_coord(coord, **self._interp_kwargs)
+        val = np.clip(val, 0, a_max=None)
+
+        return u.Quantity(val, self.map.unit, copy=False)
+
+    def write(self, overwrite=False):
+        if self.filename is None:
+            raise IOError("Missing filename")
+        elif os.path.isfile(self.filename) and not overwrite:
+            log.warning("Template file already exits, and overwrite is False")
+        else:
+            self.map.write(self.filename)
+
+    @classmethod
+    def from_dict(cls, data):
+        data = data["spatial"]
+        filename = data["filename"]
+        m = WcsNDMap.read(filename)
```

Comment:
Can't there be other types? e.g. `HpxNDMap`? You could use `Map.read` instead.

Otherwise you can create a model with such a map, write it but not read it back.